src/main/kotlin/no/nav/familie/ef/sak/økonomi/ØkonomiUtils.kt

```diff
@@ -68,57 +53,49 @@ object ØkonomiUtils {
      * @param[beståendeAndelerIHverKjede] andeler man må bygge opp etter
      * @return andeler som må bygges fordelt på kjeder
      */
-    fun andelerTilOpprettelse(oppdaterteKjeder: Map<KjedeId, List<AndelTilkjentYtelse>>,
-                              beståendeAndelerIHverKjede: Map<KjedeId, List<AndelTilkjentYtelse>>)
-            : Map<KjedeId, List<AndelTilkjentYtelse>> {
-
-        val sisteBeståendeAndelIHverKjede =
-                beståendeAndelerIHverKjede.map { (kjedeId, kjede) ->
-                    Pair(kjedeId, kjede.sortedBy { it.periodeId }.lastOrNull())
-                }.toMap()
-
-        return oppdaterteKjeder.map { (kjedeId, oppdaterteAndeler) ->
-            if (sisteBeståendeAndelIHverKjede[kjedeId] != null)
-                Pair(kjedeId, oppdaterteAndeler
-                        .filter { it.stønadFom.isAfter(sisteBeståendeAndelIHverKjede[kjedeId]!!.stønadTom) })
-            else Pair(kjedeId, oppdaterteAndeler)
+    fun andelerTilOpprettelse(oppdaterteKjeder: List<AndelTilkjentYtelse>,
+                              beståendeAndelerIHverKjede: List<AndelTilkjentYtelse>)
+            : List<AndelTilkjentYtelse> {
+
+        val sisteBeståendeAndelIHverKjede = beståendeAndelerIHverKjede.sortedBy { it.periodeId }.lastOrNull()
+
+        return if (sisteBeståendeAndelIHverKjede !== null) {
```

Comment:
Gillar också let,apply, also så denna delen hade JAG skrivit om til noe sånt ...)
return beståendeAndelerIHverKjede.sortedBy { it.periodeId }.lastOrNull()
                       ?.let { oppdaterteKjeder.filter { it.stønadFom.isAfter(it.stønadTom) } }
               ?: oppdaterteKjeder